### ISSUE TITLE
Return Bodies button doesn't work correctly after bodies collide

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/my-solar-system#139]
**Test device**
MacBook Air M1 chip

**Operating System**
13.2.1

**Browser**
Safari 

**Problem description**
For https://github.com/phetsims/qa/issues/927, if the yellow and pink bodies collide, causing the pink body to move offscreen, pressing the Return Bodies button only returns a yellow colored body in the pink body starting position.

**Steps to reproduce**
1. On either screen, increase the mass of the pink body to 300
2. Move the 2 bodies closer so they will collide
3. Press play 

**Visuals**

https://user-images.githubusercontent.com/87318828/230393949-132bf204-2eea-483b-9441-0cfdb7ef0fed.mp4

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪My Solar System‬
URL: https://phet-dev.colorado.edu/html/my-solar-system/1.1.0-dev.9/phet/my-solar-system_all_phet.html
Version: 1.1.0-dev.9 2023-04-05 16:58:00 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36
Language: en-US
Window: 1435x773
Pixel Ratio: 1.7999999523162842/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}
</details>

**Comment by samreid** [comment]
Some comments based on the changes above:

1. I tested after pulling the changes, and saw that the bug still behaves the same. The yellow body does not return when you press "return bodies"
2. `activeIndeces` should be changed to `activeIndices`
3. It is unfortunate to have to specify `index` explicitly in each item, and that could lead to maintainability copy/paste problems.  Is there a way to infer the index from the location in the array?

**Comment by AgustinVallejo** [comment]
Hi! Thanks for the feedback.

1. Right now it's behaving as expected: The pink body destroys the yellow and then flies offscreen, right now the design says that 'Return bodies' only brings back the ones that go offscreen, the same way Gravity and Orbit does. In the original issue, the pink was being returned but it came back as yellow. Now it comes just as pink and yellow should not come back.
2. Noted, will fix.
3. We could pair to discuss this... This problem was precisely arising from infering the index from location. Say you destroy bodies 0 (yellow) and 1 (pink), only 2 and 3 remain. If you infer index from array position, [2, 3] could potentially become [0, 1].

**Comment by oliver-phet** [comment]
@AgustinVallejo I'm reopening this issue. A user wrote in about this and this still seems to be problematic?

> Hello, my name is .... and I helped you troubleshoot the new browser version of the My Solar System sim. I recently encountered a new problem while rewriting an activity for the new version. In one activity, students "drop" planets of various masses into a star with a much greater mass and notice that the time to fall is independent of the dropped planet's mass. Unfortunately, in the new sim when the planet collides with the star it disappears and so does its entry row in the data table. Students would have to reenter all the data for each trial.
> 
> There is a "Return Bodies" button that appears after the collision. I presume clicking on this should return the planet and its row in the data table. This feature does not appear to be working as intended. Here is a sequence of screenshots, I also attached a short movie of what is happening.
> 
> Initial setup
> <img width="1451" alt="Image" src="https://github.com/user-attachments/assets/781496cc-8f7b-485e-9566-701dc90a79e2" />
> 
> After collision:
> <img width="1451" alt="Image" src="https://github.com/user-attachments/assets/50ddad3d-cbe1-4aab-8e6f-2e41b1863fd7" />
> 
> After clicking on Return Bodies
>  <img width="1451" alt="Image" src="https://github.com/user-attachments/assets/24902621-b2ce-4fa7-b36b-55233e639f06" />
> 
> As you can see, the body did not return. What is puzzling is that when this first happened, I kept trying different approaches and it started to work! The row of data and the body returned. I am not sure what did this but when I started again, it stopped working and I have not got it to work since. I tried Firefox, Safari, and Chrome, they all did the same thing.
> 
> I would appreciate it if you could look into this and advise me on how to get Return Bodies to work or fix it if it is a glitch. The activity is intended for 8-9th grade Earth science and astronomy students. I will post it to the PhET website once it is finished.

I read through some related issues like https://github.com/phetsims/my-solar-system/issues/147 and it seems the behavior the user is desiring was intended to be supported? The teacher tips also indicate support?

> • The Return Bodies button appears when a body escapes the screen or a natural
> collision occurs.
> • If a body is destroyed or escapes, the sim will no longer save new states. Then the
> buttons will return the bodies to their last values introduced by the user before a body
> is destroyed or scape.

**Comment by AgustinVallejo** [comment]
You're correct, this was broken on this commit https://github.com/phetsims/my-solar-system/commit/18cc9b349467a7d6e1f1fe3164ae4381e07b078e as the collided objects went into reset, and changing the mass triggered a state save. Fixing in the above commit. Will do a MR shortly

**Comment by AgustinVallejo** [comment]
@oliver-phet This is now fixed in production, you can check it at https://phet.colorado.edu/sims/html/my-solar-system/latest/my-solar-system_all.html?locale=es. Feel free to notify the user and close

**Comment by oliver-phet** [comment]
Email to user sent, closing!
